From a835ee279bf88650c0eba0442f08bbc86620aac3 Mon Sep 17 00:00:00 2001
From: Ian Stokes <ian.stokes@intel.com>
Date: Tue, 26 Nov 2019 12:03:29 +0000
Subject: dpdk: Use DPDK 18.11.5 release.

Modify travis linux build script to use the latest DPDK stable release
18.11.5. Update docs for latest DPDK stable releases.

Signed-off-by: Ian Stokes <ian.stokes@intel.com>
Acked-by: Ilya Maximets <i.maximets@ovn.org>
Acked-by: Kevin Traynor <ktraynor@redhat.com>
---
 .travis/linux-build.sh                   | 2 +-
 Documentation/faq/releases.rst           | 4 ++--
 Documentation/intro/install/dpdk.rst     | 8 ++++----
 Documentation/topics/dpdk/vhost-user.rst | 6 +++---
 NEWS                                     | 3 +++
 5 files changed, 13 insertions(+), 10 deletions(-)

diff --git a/.travis/linux-build.sh b/.travis/linux-build.sh
index 17428fa6b..1afa42914 100755
--- a/.travis/linux-build.sh
+++ b/.travis/linux-build.sh
@@ -105,7 +105,7 @@ fi
 
 if [ "$DPDK" ] || [ "$DPDK_SHARED" ]; then
     if [ -z "$DPDK_VER" ]; then
-        DPDK_VER="18.11.2"
+        DPDK_VER="18.11.5"
     fi
     install_dpdk $DPDK_VER
     if [ "$CC" = "clang" ]; then
diff --git a/Documentation/faq/releases.rst b/Documentation/faq/releases.rst
index e18f5db75..cc06571f5 100644
--- a/Documentation/faq/releases.rst
+++ b/Documentation/faq/releases.rst
@@ -178,8 +178,8 @@ Q: What DPDK version does each Open vSwitch release work with?
     2.8.x        17.05.2
     2.9.x        17.11.4
     2.10.x       17.11.4
-    2.11.x       18.11.2
-    2.12.x       18.11.2
+    2.11.x       18.11.5
+    2.12.x       18.11.5
     ============ =======
 
 Q: Are all the DPDK releases that OVS versions work with maintained?
diff --git a/Documentation/intro/install/dpdk.rst b/Documentation/intro/install/dpdk.rst
index 6e5f1ea60..13aa8a16f 100644
--- a/Documentation/intro/install/dpdk.rst
+++ b/Documentation/intro/install/dpdk.rst
@@ -42,7 +42,7 @@ Build requirements
 In addition to the requirements described in :doc:`general`, building Open
 vSwitch with DPDK will require the following:
 
-- DPDK 18.11.2
+- DPDK 18.11.5
 
 - A `DPDK supported NIC`_
 
@@ -71,9 +71,9 @@ Install DPDK
 #. Download the `DPDK sources`_, extract the file and set ``DPDK_DIR``::
 
        $ cd /usr/src/
-       $ wget http://fast.dpdk.org/rel/dpdk-18.11.2.tar.xz
-       $ tar xf dpdk-18.11.2.tar.xz
-       $ export DPDK_DIR=/usr/src/dpdk-stable-18.11.2
+       $ wget http://fast.dpdk.org/rel/dpdk-18.11.5.tar.xz
+       $ tar xf dpdk-18.11.5.tar.xz
+       $ export DPDK_DIR=/usr/src/dpdk-stable-18.11.5
        $ cd $DPDK_DIR
 
 #. (Optional) Configure DPDK as a shared library
diff --git a/Documentation/topics/dpdk/vhost-user.rst b/Documentation/topics/dpdk/vhost-user.rst
index fab87bd3d..a0d35cdd4 100644
--- a/Documentation/topics/dpdk/vhost-user.rst
+++ b/Documentation/topics/dpdk/vhost-user.rst
@@ -392,9 +392,9 @@ To begin, instantiate a guest as described in :ref:`dpdk-vhost-user` or
 DPDK sources to VM and build DPDK::
 
     $ cd /root/dpdk/
-    $ wget http://fast.dpdk.org/rel/dpdk-18.11.2.tar.xz
-    $ tar xf dpdk-18.11.2.tar.xz
-    $ export DPDK_DIR=/root/dpdk/dpdk-stable-18.11.2
+    $ wget http://fast.dpdk.org/rel/dpdk-18.11.5.tar.xz
+    $ tar xf dpdk-18.11.5.tar.xz
+    $ export DPDK_DIR=/root/dpdk/dpdk-stable-18.11.5
     $ export DPDK_TARGET=x86_64-native-linuxapp-gcc
     $ export DPDK_BUILD=$DPDK_DIR/$DPDK_TARGET
     $ cd $DPDK_DIR
diff --git a/NEWS b/NEWS
index c81890b7e..b9be7f7ce 100644
--- a/NEWS
+++ b/NEWS
@@ -1,5 +1,8 @@
 v2.12.1 - xx xxx xxxx
 ---------------------
+   - DPDK:
+     * OVS validated with DPDK 18.11.5, due to the inclusion of a fix for
+       CVE-2019-14818, this DPDK version is strongly recommended to be used.
 
 v2.12.0 - 03 Sep 2019
 ---------------------
-- 
2.14.1


